app/services/earnings_csv_parser.rb

```diff
@@ -0,0 +1,83 @@
+require 'csv'
+require_relative '../models/earning'  # Add this line if Earning model is not being loaded
+
+class EarningsCsvParser
+  STRING = 'STRING'.freeze
+  DECIMAL = 'DECIMAL'.freeze
+  # file = CSV string
+  def initialize(file, employer_id)
+      @file = file
+      @employer = get_employer(employer_id)
+      @employer_csv_format = get_employer_csv_format
+  end
+
+  def create_earnings
+      csv = CSV.parse(@file, headers: true)
+      errors = []
+      csv.each do |row|
+        e = create_earning(row)
+        if e.valid?
+          e.save
+        else
+          errors << "Error with row #{row}: #{e.errors.full_messages}"
+        end
+      end
+      display_errors errors
+  end
+
+  private
+
+  def get_employer_csv_format
+    EmployerCsvFormat.find_by!(employer_id: @employer.id)
+  end
+  
+  def get_employer employer_id
+    Employer.find_by!(id: employer_id) 
+  end
+  
+  def get_employee_id external_ref
+    Employee.find_by(external_ref: external_ref, employer_id: @employer.id)&.id
+  end
+  
+  def create_earning row
+    external_ref_header = @employer_csv_format.external_ref_header
+  
+    Earning.new(
+      earning_date: format_date(row), 
+      amount: format_amount(row), 
+      employee_id: get_employee_id(row[external_ref_header])
+    ) 
+  end
+    
+  def format_date row
```

Comment:
Make sure to have consistency with:
- 2 vs. 4 spaces
- Defining methods wrapping arguments with parentheses or not)
- Having an extra newline at the end of file

Reply:
I usually have prettier autocorrect this stuff, but I was using my personal computer and hadn't set it up. I appreciate having consistent syntax.